haystack/nodes/image_to_text/transformers.py

```diff
@@ -0,0 +1,141 @@
+        :param devices: List of torch devices (e.g. cuda, cpu, mps) to limit inference to specific devices.
+                        A list containing torch device objects and/or strings is supported (For example
+                        [torch.device('cuda:0'), "mps", "cuda:1"]). When specifying `use_gpu=False` the devices
+                        parameter is not used and a single cpu device is used for inference.
+        """
+        super().__init__()
+
+        self.devices, _ = initialize_device_settings(devices=devices, use_cuda=use_gpu, multi_gpu=False)
+        if len(self.devices) > 1:
+            logger.warning(
+                "Multiple devices are not supported in %s inference, using the first device %s.",
+                self.__class__.__name__,
+                self.devices[0],
+            )
+
+        self.model = pipeline(
+            task="image-to-text",
+            model=model_name_or_path,
+            revision=model_version,
+            device=self.devices[0],
+            use_auth_token=use_auth_token,
+        )
+        self.generate_kwargs = generate_kwargs
+        self.batch_size = batch_size
+        self.progress_bar = progress_bar
+
+    def generate_captions(
+        self, image_file_paths: List[str], generate_kwargs: Optional[dict] = None, batch_size: Optional[int] = None
+    ) -> List[Document]:
+        """
+        Generate captions for provided image files
+
+        :param image_file_paths: Paths of the images
+        :param generate_kwargs: Dictionary containing arguments for the generate method of the Hugging Face model.
+                                See https://huggingface.co/docs/transformers/en/main_classes/text_generation#transformers.GenerationMixin.generate
+        :param batch_size: Number of images to process at a time.
+        :return: List of Documents. Document.content is the caption. Document.meta["image_file_path"] contains the image file path.
+        """
+        generate_kwargs = generate_kwargs or self.generate_kwargs
+        batch_size = batch_size or self.batch_size
+
+        if len(image_file_paths) == 0:
+            raise AttributeError("ImageToText needs at least one filepath to produce a caption.")
+
+        images_dataset = ListDataset(image_file_paths)
+
+        captions: List[str] = []
+
+        for captions_batch in tqdm(
+            self.model(images_dataset, generate_kwargs=generate_kwargs, batch_size=batch_size),
+            disable=not self.progress_bar,
+            total=len(images_dataset),
+            desc="Generating captions",
+        ):
+            captions.append("".join([el["generated_text"] for el in captions_batch]).strip())
+
+        result: List[Document] = []
+        for caption, image_file_path in zip(captions, image_file_paths):
+            document = Document(content=caption, content_type="text", meta={"image_file_path": image_file_path})
```

Comment:
Can be simply `image_path` for brevity

```suggestion
            document = Document(content=caption, content_type="text", meta={"image_path": image_file_path})
```